module/validation/receipt_validator_test.go

```diff
@@ -850,3 +925,222 @@ func (s *ReceiptValidationSuite) TestValidateReceiptResultHasEnoughReceipts() {
+	candidate := unittest.BlockWithParentFixture(unknownExecutedBlock.Header)
+	candidate.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(r)))
+	err = s.receiptValidator.ValidatePayload(candidate)
+	s.Require().Error(err, "should reject invalid receipt")
+	s.Assert().True(module.IsUnknownBlockError(err), err)
+	s.Assert().False(engine.IsInvalidInputError(err), err)
+}
+
+// TestException_SealsHighestInFork tests that unexpected exceptions raised by the dependency
+// `receiptValidator.headers.ByBlockID(..)` are escalated and not misinterpreted as
+// `InvalidInputError` or `UnknownBlockError` or `UnknownResultError`
+func (s *ReceiptValidationSuite) TestException_HeadersByBlockID() {
+	s.publicKey.On("Verify", mock.Anything, mock.Anything, mock.Anything).Return(true, nil).Maybe()
+
+	//s.T().Run("seals yields exception on Identity retrieval", func(t *testing.T) {
+	valSubgrph := s.ValidSubgraphFixture()
+	s.AddSubgraphFixtureToMempools(valSubgrph)
+
+	receipt := unittest.ExecutionReceiptFixture(unittest.WithExecutorID(s.ExeID), unittest.WithResult(valSubgrph.Result))
+	candidate := unittest.BlockWithParentFixture(valSubgrph.Block.Header)
+	candidate.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt)))
+
+	// receiptValidator.headers yields exception on retrieving any block header
+	*s.HeadersDB = mock_storage.Headers{} // receiptValidator has pointer to this field, which we override with a new state mock
+	exception := errors.New("headers.ByBlockID() exception")
+	s.HeadersDB.On("ByBlockID", mock.Anything).Return(nil, exception)
+
+	err := s.receiptValidator.ValidatePayload(candidate)
+	s.Require().Error(err, "ValidatePayload should escalate exception")
+	s.Assert().False(engine.IsInvalidInputError(err), err)
+	s.Assert().False(module.IsUnknownBlockError(err), err)
+	s.Assert().False(module.IsUnknownResultError(err), err)
+}
+
+// TestException_SealsHighestInFork tests that unexpected exceptions raised by the dependency
+// `receiptValidator.seals.HighestInFork(..)` are escalated and not misinterpreted as
+// `InvalidInputError` or `UnknownBlockError` or `UnknownResultError`
+func (s *ReceiptValidationSuite) TestException_SealsHighestInFork() {
+	s.publicKey.On("Verify", mock.Anything, mock.Anything, mock.Anything).Return(true, nil).Maybe()
+	valSubgrph := s.ValidSubgraphFixture()
+	s.AddSubgraphFixtureToMempools(valSubgrph)
+
+	receipt := unittest.ExecutionReceiptFixture(unittest.WithExecutorID(s.ExeID), unittest.WithResult(valSubgrph.Result))
+	candidate := unittest.BlockWithParentFixture(valSubgrph.Block.Header)
+	candidate.SetPayload(unittest.PayloadFixture(unittest.WithReceipts(receipt)))
+
+	// receiptValidator.seals yields exception on retrieving highest sealed block in fork up to candidate's parent
+	*s.SealsDB = mock_storage.Seals{} // receiptValidator has pointer to this field, which we override with a new state mock
```

Comment:
```suggestion
	*s.SealsDB = *mock_storage.NewSeals(s.T()) // receiptValidator has pointer to this field, which we override with a new state mock
```
Would advice to use `NewSeals` to make sure that call will be indeed executed.

Reply:
👍 done in https://github.com/onflow/flow-go/pull/5211/commits/7623a680e8ca41258f7451aaa6db229da2efa935